src/ledger/events.rs

```diff
@@ -0,0 +1,95 @@
+use chrono::DateTime;
+use chrono::Utc;
+use ethereum_types::U256;
+use revm::primitives::Address;
+use uuid::Uuid;
+
+use crate::eth::primitives::BlockNumber;
+use crate::eth::primitives::Hash;
+
+/// Represents token transfers (debits and credits) associated with a specific Ethereum account within a single transaction.
+///
+/// The `account_address` field identifies the main account involved in these transfers.
+///
+/// An event will be generated for each account involved in a transaction, meaning if a transaction is not the main account in this event,
+/// in another event it will be and credit and debit parties will be adjusted accordingly.
+pub struct AccountTransfersEvent {
+    /// ID of the event publication.
+    ///
+    /// If the event is republished, a new ID will be generated for the publication.
+    ///
+    /// Format: UUID v7 (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx)
+    pub publication_id: Uuid,
+
+    /// Datetime of the event publication.
+    ///
+    /// Format: ISO 8601
+    pub publication_datetime: DateTime<Utc>,
+
+    /// Idempotency key of the event payload (TBD: `account_address` + `transaction_hash`).
+    ///
+    /// It is unique for each distinct event, but consistent across retries or republications of the same event payload.
+    ///
+    /// Format: String (TBD)
+    pub idempotency_key: String,
+
+    /// Address of the account that is part of all transfers.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub account_address: Address,
+
+    /// Hash of the Ethereum transaction that originated transfers.
+    ///
+    /// Format: Prefixed transaction hash - 32 bytes - 0x1234567890123456789012345678901234567890123456789012345678901234
+    pub transaction_hash: Hash,
+
+    /// Address of the contract that originated transfers.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub contract_address: Address,
+
+    /// Identifier of the Ethereum function that originated transfers.
+    ///
+    /// Format: Prefixed function signature - 4 bytes - 0x12345678
+    pub function_id: [u8; 4],
+
+    /// Number of the block that originated transfers.
+    ///
+    /// Format: Integer
+    pub block_number: BlockNumber,
+
+    /// Datetime of the block that originated account transfers.
+    ///
+    /// Format: ISO 8601
+    pub block_datetime: DateTime<Utc>,
+
+    /// List of transfers the related `account_address` is part of.
+    pub transfers: Vec<AccountTransfer>,
```

Comment:
If necessary, this can be split in two arrays (credits and debits).

`credits` would contains only transactions where the account_address is the credit party and `debits` where it is the debit party.

But the current approach with a single array may work better if expanding the array in multiple rows (unnest operation).

Reply:
I think single array is the way to go.

Reply:
Just as a clarification, `account_address` here is the `primary_account_address`? 
If so I think we should decide on one name to use throughout the documentation/code as to avoid any confusion.

Reply:
@carneiro-cw Yes, `account_address` was the original name before I renamed it to `primary_account_address`. This was just a comment I missed renaming. Now it is fixed.